app/_hub/kong-inc/grpc-web/index.md

```diff
@@ -0,0 +1,158 @@
+  route_id: true
+  protocols: ["http", "https"]
+  dbless_compatible: true
+
+  config:
+    - name: proto
+      required: false
+      default:
+      value_in_examples: path/to/hello.proto
+      description: |
+        If present, describes the gRPC types and methods.
+        Required to support payload transcoding.  When absent, the
+        web client must use application/grpw-web+proto content.
+
+---
+
+## Purpose
+
+A service that presents a gRPC API can be used by clients written in many languages, but the network specifications are oriented primarily to connections within a datacenter.  In order to expose the API to the Internet, and to be called from brower-based JS apps, [gRPC-Web] was developed.
+
+This plugin translates requests and responses between gRPC-Web and "real" gRPC.  Supports both HTTP/1.1 and HTTP/2, over plaintext (HTTP) and TLS (HTTPS) connections.
+
+## Usage
+
+This plugin is intended to be used in a Kong route between a gRPC service and an HTTP endpoint.
+
+Sample configuration via declarative (YAML):
+
+```yaml
+_format_version: "1.1"
+services:
+- protocol: grpc
+  host: localhost
+  port: 9000
+  routes:
+  - protocols:
+    - http
+    paths:
+    - /
+    plugins:
+    - name: grpc-web
+```
+
+Same thing via the administation API:
+
+```bash
+$ # add the gRPC service
+$ curl -XPOST localhost:8001/services \
+  --data name=grpc \
+  --data protocol=grpc \
+  --data host=localhost \
+  --data port=9000
+
+$ # add an http route
+$ curl -XPOST localhost:8001/services/grpc/routes \
+  --data protocols=http \
+  --data name=web-service \
+  --data paths=/
+
+$ # add the plugin to the route
+$ curl -XPOST localhost:8001/routes/web-service/plugins \
+  --data name=grpc-web
+```
+
+In these examples we don't set any configuration for the plugin.  This minimal setup works for the default varieties of the [gRPC-Web protocol], which use ProtocolBuffer messages either directly in binary or with base64 encoding.  The related `Content-Type` headers are `application/grpc-web` or `application/grpc-web+proto` for binary and `application/grpc-web-text` or `application/grpc-web-text+proto`.
+
+If we wish to use JSON encoding, we have to provide the gRPC specification in a .proto file.  For example:
+
+```protobuf
+syntax = "proto2";
+
+package hello;
+
+service HelloService {
+  rpc SayHello(HelloRequest) returns (HelloResponse);
+  rpc LotsOfReplies(HelloRequest) returns (stream HelloResponse);
+  rpc LotsOfGreetings(stream HelloRequest) returns (HelloResponse);
+  rpc BidiHello(stream HelloRequest) returns (stream HelloResponse);
+}
+
+message HelloRequest {
+  optional string greeting = 1;
+}
+
+message HelloResponse {
+  required string reply = 1;
+}
+```
+
+The declarative configuration becomes:
+
+```yaml
+_format_version: "1.1"
+services:
+- protocol: grpc
+  host: localhost
+  port: 9000
+  routes:
+  - protocols:
+    - http
+    paths:
+    - /
+    plugins:
+    - name: grpc-web
+      proto: path/to/hello.proto
```

Comment:
A question that popped to my mind, which could probably be answered in the documentation: does the path need to be absolute or can in be relative? (if so, relative to what?)

Reply:
good point.  I'll add some path handling code to make it similar to what other plugins assume

Reply:
very few examples of paths in configs, but the code already does the (almost) obvious thing: the current directory is the Kong prefix (`/usr/local/kong/' by default).   added to doc